### ISSUE TITLE
Error: undefined variable "data" in `check` block

### ISSUE THREAD

**Issue by ya-netgrp** [anyalir/behave#4]
Documentation claims: 
>`check`s cannot store any values. Their return values are always discarded. However, **they have access to both the `data` and `results` maps**

https://github.com/anyalir/behave/blob/34f72a5f3893c02e745e19a6e023146ac919081f/README.md?plain=1#L118

`elixir: 1.15.4-otp-26`:

```elixir
defmodule TestPhxProjWeb.AccountsTest do
  use ExUnit.Case
  use Behave, steps: [TestPhxProjTest.Web.AccountsSteps]

  @moduletag :acceptance

  scenario "login existing user" do
    given"signed up user"
    act "signs in with"
    check "received session token"
  end
end

```

```elixir
defmodule TestPhxProjTest.Web.AccountsSteps do
  use Behave.Scenario
  use TestPhxProjTest.Web.ConnCase

  given "signed up user" do
    {:user, :test}
  end

  act "signs in with" do
    t = if data.user == :test, do: "123", else: "321"
    {:token, t}
  end

  check "received session token" do
    assert data.user
    assert results.token
  end
end

```

```bash
$ MIX_ENV=test mix test
Compiling 1 file (.ex)
error: undefined variable "data"
  test/support/test_phx_proj_test/web/accounts_steps.ex:15: TestPhxProjTest.Web.AccountsSteps.check_received_session_token/2


== Compilation error in file test/support/test_phx_proj_test/web/accounts_steps.ex ==
** (CompileError) test/support/test_phx_proj_test/web/accounts_steps.ex: cannot compile module TestPhxProjTest.Web.AccountsSteps (errors have been logged)
```

**Comment by anyalir** [comment]
Yes this is a bug. One of the clauses of the act macro doesn't inject the data map.

**Comment by anyalir** [comment]
Fixed in 0.2.0. The data maps are no longer just injected in the macro, they need to be actively dereferenced as an argument to a step :)
